pykokkos/interface/type_inference.py

```diff
@@ -0,0 +1,234 @@
+                    updated_types.is_arg.add(param.name)
+
+            elif isinstance(handled_args.policy, TeamPolicy):
+                if i == 0:
+                    updated_types.inferred_types[param.name] = 'pk.TeamMember'
+                    updated_types.is_arg.add(param.name)
+
+            elif isinstance(handled_args.policy, MDRangePolicy):
+                total_dims = len(handled_args.policy.begin) 
+                if i < total_dims:
+                    updated_types.inferred_types[param.name] = "int"
+                    updated_types.is_arg.add(param.name)
+            else:
+                raise ValueError("Automatic annotations not supported for this policy")
+
+            # last policy param for parallel reduce and second last for parallel_scan is always the accumulator; the default type is double
+            if i == policy_params - 1 and parallel_type == "parallel_reduce" or i == policy_params - 2 and parallel_type == "parallel_scan":
+                updated_types.inferred_types[param.name] = "Acc:double"
+                updated_types.is_arg.add(param.name)
+
+            if i == policy_params - 1 and parallel_type == "parallel_scan":
+                updated_types.inferred_types[param.name] = "bool"
+                updated_types.is_arg.add(param.name)
+
+    if len(param_list) == policy_params:
+        if not len(updated_types.inferred_types): return None
+        return updated_types
+
+    # Handle Keyword args, make sure they are treated by queuing them in args
+    if len(passed_kwargs.keys()):
+        # add value to arguments list so the value can be assessed
+        for param in param_list[policy_params:]:
+            if param.name in passed_kwargs:
+                args_list.append(passed_kwargs[param.name])
+
+    # Handling arguments other than policy args, they begin at value_idx in args list
+    value_idx: int = 3 if handled_args.name != None else 2 
+
+    assert (len(param_list) - policy_params) == len(args_list) - value_idx, f"Unannotated arguments mismatch {len(param_list) - policy_params} != {len(args_list) - value_idx}"
+
+    # At this point there must more arguments to the workunit that may not have their types annotated
+    # These parameters may also not have raw values associated in the stand alone format -> infer types from the argument list
+
+    # DataType class has all supported pk datatypes, we ignore class members starting with __
+    supported_np_dtypes = [attr for attr in dir(DataType) if not attr.startswith("__")]
+
+    for i in range(policy_params , len(param_list)):
+        param = param_list[i]
+        value = args_list[value_idx + i - policy_params]
+
+        if isinstance(value, View) and value.layout != Layout.LayoutDefault:
+            updated_types.layout_change[param.name] = "LayoutRight" if value.layout == Layout.LayoutRight else "LayoutLeft"
+
+        if param.annotation is not inspect._empty:
```

Comment:
This should probably come before the previous if statement right?

Reply:
No because we allow mix and match. If a view has annotations but is missing layout decorator, the decorator won't be corrected. Hence we check for layout first regardless, and then move to checking annotations

Reply:
Please resolve this if you think the explanation suffices